src/components/Planner/Planner.tsx

```diff
@@ -91,12 +91,16 @@ const Planner = () => {
 
   const [isHovering, setIsHovering] = useState<HoveringButtonState>({});
 
-  const onMouseEnter = (mealId: string) => {
-    setIsHovering({ ...isHovering, [mealId]: true });
+  const onMouseEnter = (mealId: string | null) => {
+    if (mealId) {
+      setIsHovering({ ...isHovering, [mealId]: true });
+    }
   };
 
-  const onMouseLeave = (mealId: string) => {
-    setIsHovering({ ...isHovering, [mealId]: false });
+  const onMouseLeave = (mealId: string | null) => {
+    if (mealId) {
+      setIsHovering({ ...isHovering, [mealId]: false });
+    }
```

Comment:
These are for the delete button which should only show up on day meals (not saved recipe meals) and day meals should always have a meal id, so this logic works fine as a workaround. It's easier than checking for `mealId` while trying to render. Also, you can't delete a meal if it doesn't have a mealId, so this still ensures that the user don't have a delete option on something that can't be deleted.

Reply:
I still have this problem 
<img width="879" alt="image" src="https://github.com/Code-the-Dream-School/dd-prac-team1-front/assets/106414157/3e2804f3-8a31-4cec-9951-934ee7db2861">

Reply:
I fixed it in the dev/search-in-planner branch